src/switch_core_media.c

```diff
@@ -7024,6 +7036,9 @@ SWITCH_DECLARE(uint8_t) switch_core_media_negotiate_sdp(switch_core_session_t *s
 				if (a_engine->read_impl.iananame) {
 					int z = 0;
 					a_engine->reset_codec = 1;
+					if (switch_core_codec_ready(&a_engine->read_codec)) {
+						switch_set_flag(&a_engine->read_codec, SWITCH_CODEC_FLAG_RESET_PENDING);
+					}
 					if (changed_pt) {
 						switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(session), SWITCH_LOG_INFO, "Force reset codec for %s\n", a_engine->read_impl.iananame);
 						if (switch_core_media_set_codec(session, 2, smh->mparams->codec_flags) != SWITCH_STATUS_SUCCESS) {
```

Comment:
If this return success then should we clear the `SWITCH_CODEC_FLAG_RESET_PENDING`?

Reply:
Good point. I think we should also reset `reset_codec` value.

Reply:
Done